backend/src/main/scala/cromwell/backend/standard/StandardAsyncExecutionActor.scala

```diff
@@ -954,6 +954,39 @@ trait StandardAsyncExecutionActor
     }
   }
 
+  /**
+   * Merge key-value pairs from previous job execution attempt with incoming pairs from backend and store them in
+   * the database. In case when there are key-value pairs with the same key name among those from previous attempt and
+   * coming from backend, backend values have higher precedence.
+   * `FailedRetryCountKey` is handled separately outside of this method 
+   *
+   */
+  private def saveKvPairsForNextAttempt(backendExecutionStatus: Future[ExecutionHandle]): Future[ExecutionHandle] = {
+    val kvsFromPreviousAttempt = jobDescriptor.prefetchedKvStoreEntries
+      .collect {
+        // We exclude FailedRetryCountKey here since it will be incremented and saved separately, after we ensure
```

Comment:
I originally envisioned that we would pass all of these values up to the engine - and allow _the engine_ to decide whether to write all of them, or none of them, depending on whether it restarts the job or not. Does that seem sensible to you?

Regardless, I think that if `FailedRetryCountKey` is now an engine concept (and I think it is...), that we should move the value out of the `backend` package and into the `engine` package - so that it's (hopefully) impossible for backends to try to write this in the first place.